src/puppetlabs/puppetdb/command.clj

```diff
@@ -81,6 +81,10 @@
   [bytes-message]
   (parse-command (update-in bytes-message [:body] #(String. % "UTF-8"))))
 
+(defn process-payload
+  [p]
+  (if (:values p) (assoc (json/dash-keys (dissoc p :values)) :values (:values p)) (json/dash-keys p)))
```

Comment:
Do you know when this has :values and when it doesn't? Seems like if we need this, it would be less confusing to use update-in and get the 2nd and 3rd arguments to if on their own line.